homeassistant/components/minecraft_server/__init__.py

```diff
@@ -0,0 +1,335 @@
+        except IOError as error:
+            _LOGGER.debug("Error occurred while trying to ping the server: %s", error)
+            self._server_online = False
+
+    async def async_update(self, now: datetime = None) -> None:
+        """Get server data from 3rd party library and update properties."""
+        # Check connection status.
+        server_online_old = self.online
+        await self.async_check_connection()
+        server_online = self.online
+
+        # Inform user once about connection state changes if necessary.
+        if (server_online_old is True) and (server_online is False):
+            _LOGGER.warning("Connection to server lost")
+        elif (server_online_old is False) and (server_online is True):
+            _LOGGER.info("Connection to server (re-)established")
+
+        # Try to update the server data if server is online.
+        if server_online:
+            try:
+                await self._async_status_request()
+            except IOError as error:
+                _LOGGER.debug(
+                    "Error occurred while trying to update the server data: %s", error
+                )
+        else:
+            # Set all properties except description and version information to
+            # unknown until server connection is established again.
+            self._players_online = None
+            self._players_max = None
+            self._players_list = None
+            self._latency = None
+
+        # Notify sensors about new data.
+        async_dispatcher_send(self._hass, self._signal_name)
+
+        # Periodically update status.
+        self._stop_periodic_update = async_track_point_in_utc_time(
+            self._hass,
+            self.async_update,
+            utcnow() + timedelta(seconds=self._scan_interval),
+        )
+
+    async def _async_status_request(self) -> None:
+        """Request server status and update properties."""
+        try:
+            status_response = await self._hass.async_add_executor_job(
+                self._mcstatus.status, self._RETRIES_STATUS
+            )
+        except IOError:
+            _LOGGER.debug("Error while requesting server status: IOError")
+            raise IOError
+        else:
+            self._description = status_response.description["text"]
+            self._version = status_response.version.name
+            self._protocol_version = status_response.version.protocol
+            self._players_online = status_response.players.online
+            self._players_max = status_response.players.max
+            self._latency = status_response.latency
+            self._players_list = []
+            if status_response.players.sample is not None:
+                for player in status_response.players.sample:
+                    self._players_list.append(player.name)
+
+
+class MinecraftServerEntity(Entity):
+    """Representation of a Minecraft Server base entity."""
+
+    def __init__(
+        self,
+        hass: HomeAssistantType,
```

Comment:
Don't pass in `hass`. It will be set on the entity when it has been added to home assistant.

Reply:
See point 2 here:
https://developers.home-assistant.io/docs/en/creating_platform_code_review.html#4-entity